pkg/proxy/iptables/proxier.go

```diff
@@ -1099,6 +1100,18 @@ func (proxier *Proxier) syncProxyRules() {
 				// Currently we only create it for loadbalancers (#33586).
 				writeLine(natRules, append(args, "-j", string(svcXlbChain))...)
 			}
+			// When a packet arrives, it will not go through NAT table again, because it is not "the first" packet.
+			// The solution is clearing the conntrack. Known issus:
+			// https://github.com/docker/docker/issues/8795
+			// https://github.com/kubernetes/kubernetes/issues/31983
+			if lp.protocol == "udp" {
+				glog.V(2).Infof("Deleting conntrack entries for udp connections")
+				err := proxier.execConntrackTool("-D", "-p", "udp", "--dport", strconv.Itoa(lp.port))
```

Comment:
This doesn't only happen once - this is going to happen periodically, unless I am mis-readng.  I think you want to do this up around line 1077 (OpenLocalPort), when you know that this is a new nodePort, rather than a refresh.  You might also do it way at the end of this function when we close nodeports we don't want any more.

But even that is not 100% correct.  I think there's a bug here where, if the same nodeport were to be release and reused inside a single refresh loop it would fail.  I think that will self-repair, though.